layers.go

```diff
@@ -106,41 +106,35 @@ func (bd *bpLayersDir) findLayers(f func(layer bpLayer) bool) []bpLayer {
 	return selectedLayers
 }
 
-func (bd *bpLayersDir) newBPLayer(name string) *bpLayer {
+func (bd *bpLayersDir) newBPLayer(name, buildpackAPI string) *bpLayer {
 	return &bpLayer{
-		layer{
+		layer: layer{
 			path:       filepath.Join(bd.path, name),
 			identifier: fmt.Sprintf("%s:%s", bd.buildpack.ID, name),
 		},
+		API: buildpackAPI,
 	}
 }
 
 type bpLayer struct {
 	layer
+	API string
 }
 
 func (bp *bpLayer) read() (platform.BuildpackLayerMetadata, error) {
-	var data platform.BuildpackLayerMetadata
 	tomlPath := bp.path + ".toml"
-	fh, err := os.Open(tomlPath)
-	if os.IsNotExist(err) {
-		return platform.BuildpackLayerMetadata{}, nil
-	} else if err != nil {
-		return platform.BuildpackLayerMetadata{}, err
-	}
-	defer fh.Close()
-	if _, err := toml.DecodeFile(tomlPath, &data); err != nil {
+	layerMetadataFile, err := buildpack.DecodeLayerMetadataFile(tomlPath, bp.API)
+	if err != nil {
 		return platform.BuildpackLayerMetadata{}, err
 	}
 	sha, err := ioutil.ReadFile(bp.path + ".sha")
 	if err != nil {
 		if os.IsNotExist(err) {
-			return data, nil
+			return platform.BuildpackLayerMetadata{LayerMetadata: platform.LayerMetadata{SHA: ""}, LayerMetadataFile: layerMetadataFile}, nil
```

Comment:
Won't `SHA` already be `""`?

Reply:
Yes it is, I just thought to make it clearer because a few lines after, we set it to `string(sha)`.
I'll remove it if you don't think it's necessary.
I'm good either way.

Reply:
I'm fine either way, just making sure I understood. If it is for clarity, I think that is fine